libs/expo/shared/ui-components/src/lib/Snackbar/Snackbar.tsx

```diff
@@ -0,0 +1,63 @@
+import { Colors } from '@monorepo/expo/shared/static';
+import { ReactElement } from 'react';
+import { StyleProp, ViewStyle } from 'react-native';
+import { Snackbar as RnpSnackbar } from 'react-native-paper';
+
+export type TSnackbarType = 'default' | 'success' | 'error';
+
+export type TSnackbar = {
+  visible: boolean;
+  message: string;
+  onDismiss: () => void;
+  type?: TSnackbarType;
+  showDuration?: number;
+  onActionPress?: () => void;
+  styles?: StyleProp<ViewStyle>;
+  actionLabel?: string;
+};
+
+export function Snackbar(props: TSnackbar): ReactElement {
+  const {
+    styles,
+    message,
+    showDuration,
+    onDismiss,
+    onActionPress,
+    visible,
+    type,
+    actionLabel = 'Close',
+  } = props;
+
+  return (
+    <RnpSnackbar
+      style={[styles, getTypeStyles(type)]}
+      visible={visible}
+      duration={showDuration}
+      onDismiss={onDismiss}
+      action={{
+        label: actionLabel,
+        onPress: onActionPress,
+      }}
+    >
+      {message}
+    </RnpSnackbar>
+  );
+}
+
+function getTypeStyles(type?: TSnackbarType): StyleProp<ViewStyle> {
+  if (type === 'error') {
+    return {
+      backgroundColor: Colors.ERROR,
+    };
+  }
+
+  if (type === 'success') {
+    return {
+      backgroundColor: Colors.SUCCESS,
+    };
+  }
+
+  return {
+    backgroundColor: Colors.PRIMARY,
+  };
+}
```

Comment:
Suggested Change: 

`interface ISnackbar extends SnackbarProps {
  type?: TSnackbarType;
  onActionPress?: () => void;
  actionLabel?: string;
}

const typeStyles = {
  error: {
    backgroundColor: Colors.ERROR,
  },
  success: {
    backgroundColor: Colors.SUCCESS,
  },
  default: {
    backgroundColor: Colors.PRIMARY,
  },
};

export function Snackbar(props: ISnackbar): ReactElement {
  const { style, onActionPress, type, actionLabel = 'Close', ...rest } = props;

  return (
    <RnpSnackbar
      style={[style, typeStyles[type || 'default']]}
      action={{
        label: actionLabel,
        onPress: onActionPress,
      }}
      {...rest}
    />
  );
}`

Reply:
Updated. FYI - the reason for the original interface design was to fully separate it from the react-paper one, and only expose what we allow to control. If we swapped out for another one then `Snackbar` interface wouldn't change. But in this case probably doesn't matter